yt_dlp/extractor/downdogapp.py

```diff
@@ -0,0 +1,191 @@
+import time
+import urllib
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+
+class DownDogAppIE(InfoExtractor):
+    SUBDOMAINS = ['www', 'meditation', 'pilates', 'hiit', 'barre', 'prenatal']
+    _VALID_URL = rf'https?://({"|".join(SUBDOMAINS)})\.downdogapp\.com/play#(?P<id>[^\?]*)'
+    _LOGIN_URL_PATH = '/json/login'
+    _MANIFEST_URL_PATH = '/manifest'
+    _NETRC_MACHINE = 'downdogapp'
+
+    def _generate_device_description(
+        self,
+        device='Chrome%20135.0.0.0',
+        os='Linux',
+        timezone='Europe%2FBerlin',
+        language='ENGLISH',
+    ):
+        # Option values are ignored by server, still necessary to pass them
+        unix_timestamp = int(time.time())
+        return {
+            'deviceDescription': device,
+            'osVersion': os,
+            'timeZone': timezone,
+            'locale': 'en-US',
+            'appType': 'ORIGINAL',
+            'timestamp': unix_timestamp,
+            'languageOption': language,
+        }
+
+    def _get_token(self, hostname):
+        """Get initial token, needed to log in"""
+        post_data = self._generate_device_description()
+        response = self._request_webpage(
+            f'https://{hostname}{self._MANIFEST_URL_PATH}',
+            None,
+            data=urlencode_postdata(post_data),
+            headers={
+                'Content-Type': 'application/x-www-form-urlencoded',
+            })
+        response_data = response.read().decode('utf-8')
+        response_dict = self._parse_json(response_data, None)
```

Comment:
use the `_download_json` helper method; using `_request_webpage` directly is adding unnecessary steps and it breaks `--write-pages`/`--load-pages`